cpp/src/arrow/c/meson.build

```
@@ -15,10 +15,16 @@
 # specific language governing permissions and limitations
 # under the License.
 
+if needs_compute
+    arrow_c_bridge_deps = [arrow_compute_test_dep]
+else
+    arrow_c_bridge_deps = []
```

Do we need `arrow_test_dep` here?

```suggestion
    arrow_c_bridge_deps = [arrow_test_dep]
```

---

Ah, I want to focus on `if not needs_compute` branch here. In the branch, we want to run `bridge_test.cc` without compute support.

---

Oh I see what you mean - nice catch!